.github/workflows/build-engine.yml

```
@@ -316,83 +195,179 @@ jobs:
         uses: actions/cache/restore@v4
-        id: directml-cache-restore
-        with:
-          key: directml-cache-v1-${{ hashFiles('download/directml_url.txt') }}
-          path: download/directml
-
-      - name: <Setup> Set up DirectML dynamic Library
-        if: steps.directml-cache-restore.outputs.cache-hit != 'true' && endswith(matrix.target, '-directml')
-        run: |
-          curl -L --retry 3 --retry-delay 5 "${{ matrix.directml_url }}" -o download/directml.zip
-          mkdir -p download/directml
-
-          # extract only dlls
-          unzip download/directml.zip bin/${{ matrix.architecture }}-win/DirectML.dll -d download/directml
-          rm download/directml.zip
-          mv download/directml/bin/${{ matrix.architecture }}-win/DirectML.dll download/directml/DirectML.dll
-          rm -r download/directml/bin
-
-      - name: <Setup> Save DirectML cache
-        if: endswith(matrix.target, '-directml')
-        uses: actions/cache/save@v4
+        id: voicevox-core-cache-restore
         with:
-          key: ${{ steps.directml-cache-restore.outputs.cache-primary-key }}
-          path: download/directml
+          key: voicevox-core-${{ runner.os }}-${{ matrix.architecture }}-${{ matrix.onnxruntime_device }}-v${{ env.VOICEVOX_CORE_AND_DOWNLOADER_VERSION }}
+          path: download/core/c_api
 
       # ONNX Runtime
-      - name: <Setup> Export ONNX Runtime url to calc hash
-        run: echo "${{ matrix.onnxruntime_url }}" > download/onnxruntime_url.txt
-
       - name: <Setup> Restore cached ONNX Runtime
         uses: actions/cache/restore@v4
         id: onnxruntime-cache-restore
         with:
-          key: ${{ matrix.os }}-onnxruntime-${{ hashFiles('download/onnxruntime_url.txt') }}-v1
-          path: download/onnxruntime
+          key: onnxruntime-${{ runner.os }}-${{ matrix.architecture }}-${{ matrix.onnxruntime_device }}-core${{ env.VOICEVOX_CORE_AND_DOWNLOADER_VERSION }}
+          path: download/core/onnxruntime
 
-      - name: <Setup> Download ONNX Runtime (Windows)
-        if: steps.onnxruntime-cache-restore.outputs.cache-hit != 'true' && runner.os == 'Windows'
+      # additional libraries
+      - name: <Setup> Restore cached additional libraries
+        if: matrix.onnxruntime_device != 'cpu'
+        uses: actions/cache/restore@v4
+        id: additional-libraries-cache-restore
+        with:
+          key: additional-libraries-${{ env.VOICEVOX_ADDITIONAL_LIBRARIES_VERSION }}-${{ matrix.onnxruntime_device }}-${{ matrix.architecture }}
+          path: download/core/additional_libraries
+
+      # models
+      - name: <Setup> Restore cached models
+        uses: actions/cache/restore@v4
+        id: models-cache-restore
+        with:
+          key: models-${{ runner.os }}-${{ matrix.architecture }}-${{ matrix.onnxruntime_device }}-core${{ env.VOICEVOX_CORE_AND_DOWNLOADER_VERSION }}
+          path: download/core/models
+
+      - name: <Setup> Install Downloader
```

「インストール」ではなく「ダウンロード」を意識したほうが使いやすそう。
```suggestion
      - name: <Setup> Download Downloader
```